### PR TITLE
[LTC] Implement Banquet Guests

### PR DESCRIPTION
**Pull request by Susucre**
I did make a StaticFilters for controlled Food called "Food", used mostly in sacrifice costs.
With the next Eldraine Set we'll certaintly have more of that.

### CODE REVIEW

**Comment by Susucre** [comment]
[[Banquet Guests]]

File: Mage/src/main/java/mage/abilities/effects/common/EntersBattlefieldWithXCountersEffect.java [hunk 1]
```diff
@@ -22,16 +22,28 @@
...
+    public EntersBattlefieldWithXCountersEffect(Counter counter, int multiplier) {
         super(Outcome.BoostCreature);
         this.counter = counter;
-        staticText = "with X " + counter.getName() + " counters on it";
+        this.multiplier = multiplier;
+
+        String countText = "X";
+        if (multiplier == 2) {
+            countText = "twice " + countText;
+        }
```
Comment (Susucre) [inline Mage/src/main/java/mage/abilities/effects/common/EntersBattlefieldWithXCountersEffect.java:39]: I was not sure what to set the text for more than 2. Any idea?

Comment (xenohedron) [reply Mage/src/main/java/mage/abilities/effects/common/EntersBattlefieldWithXCountersEffect.java:39]: If multiplier > 2
`CardUtil.numberToText(multiplier) + " times " + countText`

File: Mage.Sets/src/mage/cards/b/BanquetGuests.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                        new AffinityEffect(StaticFilters.FILTER_CONTROLLED_FOOD)
+                ).addHint(hint)
+        );
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+        // Banquet Guests enters the battlefield with twice X +1/+1 counters on it.
+        this.addAbility(new EntersBattlefieldAbility(
+                new EntersBattlefieldWithXCountersEffect(CounterType.P1P1.createInstance(), 2)
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/b/BanquetGuests.java:58]: It's also possible to add a count argument to `createInstance()`. If you think the separate multiplier is cleaner then I'm fine with it.

Comment (Susucre) [reply Mage.Sets/src/mage/cards/b/BanquetGuests.java:58]: Issue is that rework would be needed inside the effect I think:

That seemed weird to me, and I do not know much about how counter replacement work:
```java
if (amount > 0) {
    Counter counterToAdd = counter.copy();
    counterToAdd.add(amount - counter.getCount());
    List<UUID> appliedEffects = (ArrayList<UUID>) this.getValue("appliedEffects");
    permanent.addCounters(counterToAdd, source.getControllerId(), source, game, appliedEffects);
}
```

Having the multiplier as parameter is easier in my opinion for static text handling.

**Review by xenohedron — APPROVED** [review]

**Pull request merged**